packages/flame/lib/src/components/clip_component.dart

```diff
@@ -144,4 +126,26 @@ class ClipComponent extends PositionComponent implements SizeProvider {
   bool containsLocalPoint(Vector2 point) {
     return _shape.containsPoint(point);
   }
+
+  /// Returns the [ShapeBuilder] function that builds a polygon
+  ///
+  /// this allows us to use an assertion during Constructor initialization
+  /// rather than at the execution of the builder function.
+  static ShapeBuilder _polygonBuilderCallback(List<Vector2> points) {
+    assert(
+      points.length >= 3,
+      'PolygonClipComponent requires at least 3 points.',
+    );
+
+    return (Vector2 size) => _polygonBuilder(points, size);
+  }
+
+  static Shape _polygonBuilder(List<Vector2> points, Vector2 size) {
+    final translatedPoints = points
+        .map(
+          (p) => p.clone()..multiply(size),
+        )
+        .toList();
+    return Polygon(translatedPoints);
+  }
```

Comment:
I would merge these two methods and call the resulting method `_polygonBuilder`, since it isn't really a callback.

Reply:
The idea behind them is, that the first gets executed during initialization and therefor the assertion is run imediatly while it then returns the builder function that is used later.

If the assertion is in the builder itself, it gets first executed in the onLoad() method if i am right.

Reply:
Right, that makes sense, the second function could still be inlined, but now when looking at it that will be a bit more expensive so probably worth having it like it is now, but coming up with a better name for  `_polygonBuilderCallback`. As always, hard with naming though. :sweat_smile:

Reply:
Maybe just `_polygonShapeBuilder`?

Reply:
I'll fix it up then, since you're in a hurry :)